packages/@tinacms/graphql/src/builder/index.ts

```diff
@@ -824,9 +868,140 @@ export class Builder {
       namespace: connectionNamespace,
       connectionName,
       nodeType: nodeType,
+      collections,
     })
   }
 
+  private _buildFieldFilter = async (field: TinaFieldEnriched) => {
```

Comment:
This is nearly identical to what we're already doing with the "mutation" params. The key difference here is that the inputs are used for building the query request, and there is an additional layer of indirection so that we can be more granular with the _type_ of request:
```graphql
query {
  getMovieList(filter: { title: { startsWith: "Star" }, archived: { eq: true } }) {
    edges {
      node {
        id
      }
    }
  }
}
```